runtime/src/main/java/org/corfudb/runtime/DistributedClientCheckpointer.java

```diff
@@ -0,0 +1,55 @@
+package org.corfudb.runtime;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Runs the checkpointing of locally opened tables from within the client's JVM
+ * <p>
+ */
+@Slf4j
+public class DistributedClientCheckpointer {
+
+    private final ScheduledExecutorService compactionScheduler;
+    private final DistributedCompactor distributedCompactor;
+
+    public DistributedClientCheckpointer(@Nonnull CorfuRuntime runtime) {
+        if (runtime.getParameters().checkpointTriggerFreqMillis <= 0) {
+            this.compactionScheduler = null;
```

Comment:
no need for null

Reply:
If not for it, distributedCompactor and compactionScheduler  cannot be made final otherwise. 
Anyway after making it null here, it returns.